src/hotspot/share/prims/jvmtiExport.cpp

```diff
@@ -1101,8 +1101,8 @@ bool JvmtiExport::post_class_file_load_hook(Symbol* h_name,
     return false;
   }
 
-  if (JavaThread::current()->is_in_VTMS_transition()) {
-    return false; // no events should be posted if thread is in VTMS transition
+  if (JavaThread::current()->hide_jvmti_events()) {
+    return false; // no events should be posted if thread is in VTMS alike transition
```

Comment:
I don't think your use of "alike" is what you mean here. Did you mean "VTMS-like", or maybe something like "VTMS transition or similar state".

Also `hide_jvmti_events()` is a pretty self explanatory name already. Probably no comment is needed at the call sites, and instead you should just add a comment where `hide_jvmti_events()` is declared.

Reply:
Thanks. Good suggestion. Will address.